TallyCore/Utility/IListExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NetTally.Utility
+{
+    public static class IListExtensions
+    {
+        public static void Swap<T>(
+            this IList<T> list,
+            int firstIndex,
+            int secondIndex
+            )
+        {
+            Contract.Requires(list != null);
```

Comment:
Also had to get rid of `Contract`, since it's not valid in .NET Standard 1.3, the target used for the core assembly when moving to VS2017 (target selected to simplify porting to Android/iOS).  Switched it to a standard argument null check.